docs/r2/migrate.ipynb

```diff
@@ -0,0 +1,404 @@
+        "colab_type": "code",
+        "id": "c50hsFk2MiWs",
+        "colab": {}
+      },
+      "source": [
+        "import tensorflow as tf"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "V_JOBTVzU5Cx"
+      },
+      "source": [
+        "## What's changed\n",
+        "\n",
+        "\n",
+        "It's useful to think of the `tf.summary` API as two sub-APIs:\n",
+        "\n",
+        "-   A set of ops for recording individual summaries - `summary.scalar()`, `summary.histogram()`, `summary.image()`, `summary.audio()`, and `summary.text()` - which are called inline from your model code.\n",
+        "-   Writing logic that collects these individual summaries and writes them to a specially formatted log file (which TensorBoard then reads to generate visualizations)."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "9-rVv-EYU8_E"
+      },
+      "source": [
+        "### In TF 1.x\n",
+        "\n",
+        "The two halves had to be manually wired together - by fetching the summary op outputs via `Session.run()` and calling `FileWriter.add_summary(output, step)`.  The `v1.summary.merge_all()` op made this easier by using a graph collection to aggregate all summary op outputs, but this approach still worked poorly for eager execution and control flow, making it especially ill-suited for TF 2.0."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "rh8R2g5FWbsQ"
+      },
+      "source": [
+        "### In TF 2.X\n",
+        "\n",
+        "A new integrated approach is used. Individual `tf.summary` ops write their data immediately when executed.  Using the API from your model code should still look familiar, but it's now friendly to eager execution while remaining graph-mode compatible.  Integrating both halves of the API means the `summary.FileWriter` is now part of the TensorFlow execution context and gets accessed directly by `tf.summary` ops, so configuring writers is the main part that looks different."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "em7GQju5VA0I"
+      },
+      "source": [
+        "\n",
+        "Example usage with eager execution, the default in TF 2.0:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "GgFXOtSeVFqP",
+        "colab": {}
+      },
+      "source": [
+        "writer = tf.summary.create_file_writer(\"/tmp/mylogs/eager\")\n",
+        "\n",
+        "with writer.as_default():\n",
+        "  for step in range(100):\n",
+        "    # other model code would go here\n",
+        "    tf.summary.scalar(\"my_metric\", 0.5, step=step)\n",
+        "    \n",
+        "writer.flush()"
```

Comment:
Can we keep this flush() call indented inside the loop?  That way it's consistent with the other examples.